### ISSUE TITLE
Possible to set up without TLS authentication?

### ISSUE THREAD

**Issue by kvanderslicesuvoda** [jenkinsci/docker-swarm-plugin#63]
Hi team,

I'm working on a small lab project and was looking at this plugin.  Until now I've been using the [docker plugin](https://plugins.jenkins.io/docker-plugin), which has been great for a single node use.

My lab does not have CA certs set up for the swarm (currently) but I wanted to evaluate the swarm plugin anyway.  Is it possible to test this without CA certificates?  Or is that a requirement of the target swarm?

Thanks!

**Comment by edumgui** [comment]
Yes, you can. Just set tcp://IP:PORT and click "Test Connection" button. It is working fine.

**Comment by gutschet** [comment]
When you publish your docker daemon without TLS on Port 2375 you can also use http://<host>.<domain>:2375 as URL

**Comment by Tumerskiy** [comment]
with 1.9 it throws NPE... with tcp:// or http://

**Comment by Roemer** [comment]
@Tumerskiy What was the last version where it worked? Can you give me a stack trace of the NPE?

**Comment by gutschet** [comment]
@Tumerskiy Did you publish the daemon sockt on http? 
On Debian, try adding a `/etc/systemd/system/docker.service.d/override.conf` with following content: 
```
[Service]
ExecStart=
ExecStart=/usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock -H tcp://0.0.0.0:2375
```
For a quick test you can also start dockerd from command line with dockerd -H=0.0.0.0:2375

**Comment by Tumerskiy** [comment]
Hi,
The behaviour of test is not consistent. Swarm is available from jenkins host by tcp 2375, and in fact, if test is ignored and config is saved for the first time, I can see that swarm appears "connected". Dashboard reflects services and apparently new service is launched once called from pipeline, need more testing though. 
Testing of the existing connection goes without errors.

**Comment by kvanderslicesuvoda** [comment]
Hi team,

I also found the behavior of the test inconsistent - once I realized that the test wasn't indicative of an actual issue, I was able to get this set up.

I have my Docker swarm set up with `daemon.json`:

```{
  "hosts": [
    "tcp://0.0.0.0:2375",
    "unix:///var/run/docker.sock"
  ],
  "experimental": true,
  "labels": ["os=linux"],
  "log-driver": "json-file",
  "log-opts": {
    "max-size": "20m",
    "max-file": "10"
  }
}`

No TLS or anything.  My mistake was using tcp:// as the Swarm manager URL.

I use Config as Code to configure my Jenkins instances, so this ends up looking like:

```
---
jenkins:
  clouds:
  - dockerSwarm:
      agentTemplates:
      - command: |-
            sh
            -cx
            curl --connect-timeout 20 --max-time 60 -o agent.jar $DOCKER_SWARM_PLUGIN_JENKINS_AGENT_JAR_URL && java -jar agent.jar -jnlpUrl $DOCKER_SWARM_PLUGIN_JENKINS_AGENT_JNLP_URL -secret $DOCKER_SWARM_PLUGIN_JENKINS_AGENT_SECRET -noReconnect -workDir /tmp
        image: dcr.myregistry.io/jenkins-executor:latest
        label: "docker"
        limitsMemoryBytes: 0
        limitsNanoCPUs: 0
        osWindows: false
        pullCredentialsId: "docker-registry"
        reservationsMemoryBytes: 0
        reservationsNanoCPUs: 0
        workingDir: "/tmp"
      dockerHost:
        uri: http://swarm.myco.io:2375
      dockerSwarmApiUrl: http://swarm.myco.io:2375
      jenkinsUrl: http://jenkins_internal_url:8080
      swarmNetwork: jenkins_swarm_network
`

Adjust accordingly to your parameters above.

**Comment by kvanderslicesuvoda** [comment]
One update is that the above CASC snippet does not work unless I manually save the config.  Afterwards, it does.

**Comment by XQDD** [comment]
+1  NPE(NULL Pointer Exception) before save configuration

![image](https://user-images.githubusercontent.com/20871761/75855609-9f74e980-5e2d-11ea-8b22-3ebdb9573154.png)
